app.rb

```diff
@@ -0,0 +1,126 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'No books available.'
+    else
+      puts 'All Books:'
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}. Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'No people available.'
+    else
+      puts 'All People:'
+      @people.each do |person|
+        puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}, Type: #{person.class}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a Student(1) or a Teacher(2)? [Input the number]: '
+    res = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    if res == 1
+      print 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp.downcase == 'y'
+      person = Student.new(age, nil, name, parent_permission: parent_permission)
+      puts "Student created successfully. ID: #{person.id}, Name: #{person.name}"
+    elsif res == 2
+      print 'Enter the teacher\'s specialization: '
+      specialization = gets.chomp
+      person = Teacher.new(nil, specialization, name: name, age: age)
+      puts "Teacher created successfully. Name: #{person.name}, ID: #{person.id}"
+    else
+      puts 'Invalid choice. Person creation failed.'
+    end
+    @people << person
+  end
+
+  def create_book
+    print 'Enter the book\'s title: '
+    title = gets.chomp
+    print 'Enter the book\'s author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  def select_person
+    puts 'Select a person to create a rental:'
+    list_all_people
+    print 'Enter the person\'s ID: '
+    person_id = gets.chomp.to_i
+    @people.find { |p| p.id == person_id }
+  end
```

Comment:
Your app shows an error when you try to create a person as shown in the image below. Kindly ensure that you give the correct number of arguments to the student class since that is the cause of the error.


![Screenshot from 2023-06-29 23-07-12](https://github.com/zamcham/OOP-school-library/assets/65233860/0a27e917-f3db-4863-9c57-0ccda65f8ac1)